catalog/app/containers/Admin/Buckets.tsx

```diff
@@ -290,125 +445,120 @@ function BucketFields({ add = false, reindex }) {
   )
 }
 
-const formToJSON = (values) => {
-  const isMissing = (v) => v == null || v === '' || Number.isNaN(v)
-  const get = (key, onValue = R.identity, onMissing) => {
-    const v = values.get(key)
-    return isMissing(v) ? onMissing : onValue(v)
-  }
-  const json = {
-    name: get('name'),
-    title: get('title', R.trim),
-    icon_url: get('iconUrl', R.identity),
-    description: get('description', R.trim),
-    relevance_score: get('relevanceScore', Number),
-    overview_url: get('overviewUrl'),
-    tags: get(
-      'tags',
-      R.pipe(
-        R.split(','),
-        R.map(R.trim),
-        R.reject((t) => !t),
-        R.uniq,
-      ),
-    ),
-    schema_org: get('linkedData', JSON.parse),
-    file_extensions_to_index: get(
-      'fileExtensionsToIndex',
-      R.pipe(
-        R.split(','),
-        R.map(R.trim),
-        R.reject((t) => !t),
-        R.uniq,
-      ),
-    ),
-    scanner_parallel_shards_depth: get('scannerParallelShardsDepth', Number),
-    sns_notification_arn: get('snsNotificationArn', (v) => {
-      if (v === DO_NOT_SUBSCRIBE_SYM) return DO_NOT_SUBSCRIBE_STR
-      if (typeof v === 'string') return v.trim()
-      return undefined
-    }),
-    skip_meta_data_indexing: get('skipMetaDataIndexing'),
-    delay_scan: get('delayScan'),
+const ADD_MUTATION = `
+  mutation Admin_Buckets_Add($input: BucketAddInput!) {
```

Comment:
Maybe better to store them in .gql files to leverage syntax highlighting?

Reply:
graphql syntax can be highlighted even inside strings (when using `gql` template tag at least), but maybe we'll move the queries out eventually for another reason -- it's easier to process these files and compile them into ts / js by webpack or maybe some other tool (inline processing is harder bc typescript doesnt support custom transformers, but it's still somewhat feasible)